azure/dns_records/variables.tf

```diff
@@ -10,30 +10,30 @@ variable "dns_zone_rg" {
 
 variable "a" {
   description = "A record to be created"
-  type = map(object({
+  type = list(object({
     name    = string
     records = list(string)
     ttl     = number
   }))
-  default = {}
+  default = []
```

Comment:
@filipefigueredo why did we change the usage from map to list here? May be its just me since i am not aware of full story.